test/pipelinerun_test.go

```diff
@@ -84,6 +85,34 @@ func TestPipelineRun(t *testing.T) {
 			}
 		}
 	}
+
+	logger.Infof("Making sure the expected events were created from taskrun and pipelinerun")
```

Comment:
you can also use https://github.com/kubernetes/client-go/blob/master/tools/record/fake.go#L54 and no need to start a timer.

Reply:
Events in integration test are not fake. I am using timer here to not wait for events until test timesout. I am not sure I understand your comment. Can you explain more?